### ISSUE TITLE
Learning rate scheduler is working incorrect

### ISSUE THREAD

**Issue by hocop** [Lornatang/SRGAN-PyTorch#1]
While training SRGAN model, learning rate scheduler is used in `train.py` with parameters `step_size=epochs // 2` and `gamma=0.1`.  
From this I suppose `schedulerD.step()` and `schedulerG.step()` to be called at each epoch.  
However, they are called at each training step which means that learning rate is quickly becoming zero after a few training steps.

The SRGAN does not seem to train.  
But when I put `schedulerD.step()` and `schedulerG.step()` after the epoch end, it started showing nice results.

**Comment by Lornatang** [comment]
Yeah, I found it. I will fix it.
